vault/policy_store.go

```diff
@@ -381,7 +391,7 @@ func (ps *PolicyStore) getBarrierView(ns *namespace.Namespace, _ PolicyType) Bar
 
 // GetPolicy is used to fetch the named policy
 func (ps *PolicyStore) GetPolicy(ctx context.Context, name string, policyType PolicyType) (*Policy, error) {
-	return ps.switchedGetPolicy(ctx, name, policyType, true)
+	return ps.switchedGetPolicy(ctx, name, policyType, false)
```

Comment:
We could leave this as `true`, but I don't understand why it is set to true in the first place; at best it should be `RLock()` rather than `Lock()` as the latter is exclusive and unnecessary. The lock is also released before returning so there's no point to it except cache<->storage consistency, which seems unnecessary. 

`switchedGetPolicy(..., false)` is not called anywhere from what I can see; everyone comes in through `GetPolicy(...)`.